drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -679,6 +700,82 @@ def has_submission_deadline_passed(self):
         else:
             return False
 
+    @property
+    def closed(self):
+        """
+        Is the student still allowed to submit answers?
+        """
+        if self.used_all_attempts:
+            return True
+        if self.has_submission_deadline_passed:
+            return True
```

Comment:
This can be used directly in `is_answer_available` instead of adding a new method. Same for `is_attempted` and `is_finished`. Unless we are using them in any other places in this XBlock, we don't gain much by introducing new properties.

Reply:
@Agrendalath 
I tried to follow the code pattern mentioned [here](https://github.com/openedx/edx-platform/blob/master/xmodule/capa_module.py#L1383-L1430) which I believe can increase the readability of ```is_answer_available```

Thanks